lib/nostrum/shard/session.ex

```diff
@@ -145,16 +145,29 @@ defmodule Nostrum.Shard.Session do
     {:next_state, :connecting_http, data}
   end
 
-  #def disconnected(_kind, _request, _data) do
+  # def disconnected(_kind, _request, _data) do
   #  {:keep_state_and_data, :postpone}
-  #end
+  # end
 
   # If we've been here before, we want to use the resume gateway URL to connect
   # instead of the regular gateway URL.
   def connecting_http(:enter, from, %{resume_gateway: resume_gateway} = data)
       when resume_gateway != nil do
     Logger.debug("Resuming on #{inspect(resume_gateway)}")
-    connecting_http(:enter, from, %{data | gateway: resume_gateway})
+
+    # strip off the wss:// prefix if it's there
+    # since :gun doesn't know how to handle it
+    resume_url =
+      case resume_gateway do
+        <<"wss://", gateway_url::binary>> ->
+          gateway_url
+
+        gateway_url ->
+          gateway_url
+      end
```

Comment:
Can't we just do

`%{resume_gateway: "wss://" <> resume_gateway}`

in the function head?